### PR TITLE
Update robo test devices MAPSAND-268

### PR DESCRIPTION
**Pull request by pengdev**
### Summary of changes

Change robo test device from virtual Pixel2 to Pixel3 and added Pixel 6 running at API level 31.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Comment by ank27** [comment]
Can we also add command to fetch model list so we know availability of devices. i.e. add `gcloud firebase test android models list
`, before running `gcloud firebase test android run ... ` ?

**Review by ank27 — APPROVED** [review]

**Review by kiryldz — COMMENTED** [review]
just an idea - perhaps simply leave following devices to save costs (and also assuming that robo tests did not catch anything useful for quite a time already):
- api 23 - differs from APIs above in terms for at least Android animators
- pixel 3, api 30 - assuming it's x86 to validate no issues there
- api 31, pretty fresh one

Other 2 devices are a bit questionable for me but in general LGTM to leave them as well

File: .circleci/config.yml [hunk 1]
```diff
@@ -326,12 +326,14 @@ commands:
           no_output_timeout: 10m
           command: |
             if [[ -n "${GCLOUD_SERVICE_ACCOUNT_JSON}" ]]; then
+              gcloud firebase test android models list
               gcloud firebase test android run --type robo \
                 --app app/build/outputs/apk/release/app-release.apk \
                 --device model=harpia,version=23,locale=en,orientation=portrait \
                 --device model=blueline,version=28,locale=en,orientation=portrait \
-                --device model=Pixel2,version=30,locale=en,orientation=portrait \
+                --device model=Pixel3,version=30,locale=en,orientation=portrait \
```
Comment (yunikkk) [inline .circleci/config.yml:334]: Drop a note that it's x86 virtual device for future context?

Comment (pengdev) [reply .circleci/config.yml:334]: I don't know if it's a x86 virtual device, but it's a virtual device, will update the note

Comment (yunikkk) [reply .circleci/config.yml:336]: Suppose we used virtual specifically because it's x86 - to test if that arch builds / runs at all

**Review by Chaoba — APPROVED** [review]

**Review by kiryldz — APPROVED** [review]

**Review by yunikkk — APPROVED** [review]

**Review by alexander-kulikovskii — APPROVED** [review]

**Comment by pengdev** [comment]
The reason behind the virtual device failure is that there’s an ongoing incident on Firebase test lab:
```
Firebase Test Lab is experiencing inclusive Robo test results running on virtual API 30 and 31. We currently estimate that 100% of users of Robo tests on virtual API 30 or 31 are affected. We are currently investigating the cause.
```
[Firebase Status Dashboard](https://status.firebase.google.com/incidents/DYUAvM5womeveV2M4B2t)

I've changed the virtual device to Pixel 2 at API level 29 to avoid the issue.

**Pull request merged**